quantecon/markov/ddp.py

```diff
@@ -456,6 +456,73 @@ def _check_action_feasibility(self):
                     'violated for state {s}'.format(s=s)
                 )
 
+    def to_sa_formulation(self, sparse=True):
+        """
+        Convert this instance of `DiscreteDP` to SA-pair form
+
+        Parameters
+        ----------
+        sparse : bool, optional(default=True)
+            Should the `Q` matrix be stored as a sparse matrix.
+            If true the CSR format is used
+
+        Returns
+        -------
+        ddp_sa : DiscreteDP
+            The correspnoding DiscreteDP instance in SA form
+
+        Notes
+        -----
+        If this instance is already in SA form then it is returned
+        un-modified
+        """
+
+        if self._sa_pair:
+            return self
+        else:
+            s_ind, a_ind = np.where(np.isfinite(self.R))
```

Comment:
Better to write `self.R > -np.inf` instead of `np.isfinite(self.R)`?
(`self.R` may contain `np.inf`.)